### PR TITLE
avc278 - 2.03 - Javascript

### PR DESCRIPTION
**Pull request by avc278**

### CODE REVIEW

File: JavaScript/chapter02/p03_delete_middle_node/avc278.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+const compareLinkedLists = (A, B) => {
+  if (!A && !B) return true;
+
+  while (A !== null && B !== null) {
+    if (A.val !== B.val) return false;
+    A = A.next;
+    B = B.next;
+  }
+  return !((A && !B) || (!A && B));
```
Comment (lhchavez) [inline JavaScript/chapter02/p03_delete_middle_node/avc278.js:32]:
```suggestion
  return A !== null && B !== nulll;
```

Comment (avc278) [reply JavaScript/chapter02/p03_delete_middle_node/avc278.js:32]: Reading back on the explanation in #65, I do think we want to return true when we have exhausted A and B.
So `return A === null && B === null` would be valid here, or just `return !A && !B`.

Comment (lhchavez) [reply JavaScript/chapter02/p03_delete_middle_node/avc278.js:32]: err... yes, i messed up the negations ^^;;

Comment (avc278) [reply JavaScript/chapter02/p03_delete_middle_node/avc278.js]: Oh btw, while we're here, would you mind if I made a separate directory in chapter 2 for re-usable functions/classes just to DRY things out?
Something like `JavaScript/chapter02/utils/avc278.js`

Comment (lhchavez) [reply JavaScript/chapter02/p03_delete_middle_node/avc278.js]: creating a directory to host re-usable functions/classes sounds good!

in some circles, the name "utils" is frowned upon (e.g. https://softwareengineering.stackexchange.com/a/175190, https://www.vojtechruzicka.com/avoid-utility-classes/), so `JavaScript/lib/avc278/linkedlist.js` may be a better location.

Comment (avc278) [reply JavaScript/chapter02/p03_delete_middle_node/avc278.js]: Oo sounds good to me. Keep an eye out for it in my next PR.
I'll also sneak in refactors to the previous solutions in chapter 2.

**Pull request merged**